components/Profile/hooks.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable consistent-return */
+import { useEffect, useState } from 'react';
+
+// Cached script statuses
+const cachedScriptStatuses = new Map();
+
+function getScriptNode(src) {
+  const node = document.querySelector(`script[src="${src}"]`);
+  const status = node?.getAttribute('data-status');
+
+  return {
+    node,
+    status,
+  };
+}
+
+export function useScript(src, options = {}) {
```

Comment:
it was actually copied from ts useHooks, but since we don't use ts I decided just to copy paste
not sure if we need the whole package, and if we're going to move this to the nx, we can simply remove this and install a package for ts, wdyt?

Reply:
lacking context, but this will probably work without the need for any further packages
https://nextjs.org/docs/pages/api-reference/components/script#onready

Reply:
next/script works well, thanks @truemiller

Reply:
actually there are difficulties with rendering <Script/> in maps, moved it out. let me know your thoughts